File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    
    public static void swap(int[] arr,int st,int ed){
        int temp=arr[st];
        arr[st]=arr[ed];
        arr[ed]=temp;
    }
    public void sortColors(int[] nums) {
        int start=0,mid=0,end=nums.length-1;
        
        while(mid<=end){
            switch(nums[mid]){

                case 0:
                  swap(nums,start,mid);
                  mid++;
                  start++;
                  break;
                case 1:
                   mid++;
                   break;
                case 2:   
                   swap(nums,mid,end);
                   end--;
                   break;
            }
        }
        
    }
}

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void swap(vector<int>& arr,int p,int q){
        int temp=arr[p];
        arr[p]=arr[q];
        arr[q]=temp;
    }
    void sortColors(vector<int>& nums) {
        int start=0, mid=0,end=nums.size()-1;

        while(mid<=end){
            switch(nums[mid]){

                case 0:
                  swap(nums,start,mid);
                  mid++;
                  start++;
                  break;
                case 1:
                  mid++;break;
                case 2:
                  swap(nums,mid,end);
                  end--;
                  break;
            }
        }
    }
};